pkg/gitparse/gitparse.go

```diff
@@ -42,6 +34,46 @@ type Diff struct {
 	IsBinary  bool
 }
 
+// Parser sets values used in GitParse.
+type Parser struct {
+	MaxDiffSize   int
+	MaxCommitSize int
+	DateFormat    string
+}
+
+// MaxDiffSize sets MaxDiffSize option. Diffs larger than MaxDiffSize will
+// be truncated.
+func MaxDiffSize(MaxDiffSize int) Option {
```

Comment:
super nit: Rename theses to `WithMaxDiffSize` and `WithMaxCommitSize` that way we are consistent with our engine and few other places we use functional variadic args.

Also the param in this function can be camelcased?
`func MaxDiffSize(MaxDiffSize int) Option` -> `func MaxDiffSize(maxDiffSize int) Option`

Same with MaxCommitSize.

I feel like making all the things private, keeps the global namespace a tiny bit cleaner.
```
func WithMaxDiffSize(maxDiffSize int) Option {
	return func(parser *Parser) {
		parser.maxDiffSize = maxDiffSize
	}
}
```